src/Xamarin.Android.Build.Tasks/Tasks/BuildArchive.cs

```diff
@@ -180,63 +195,89 @@ public override bool RunTask ()
 			if (string.Compare (Path.GetFileName (entry), "AndroidManifest.xml", StringComparison.OrdinalIgnoreCase) == 0)
 				continue;
 
-			Log.LogDebugMessage ($"Removing {entry} as it is not longer required.");
-			apk.Archive.DeleteEntry (entry);
+			Log.LogDebugMessage ($"Removing {entry} as it is no longer required.");
+			apk.DeleteEntry (entry);
 		}
 
-		if (string.Compare (AndroidPackageFormat, "aab", true) == 0)
+		if (is_aab)
 			FixupArchive (apk);
 
 		return !Log.HasLoggedErrors;
 	}
 
-	bool AddFileToArchiveIfNewer (ZipArchiveEx apk, string file, string inArchivePath, ITaskItem item, List<string> existingEntries)
+	// .NET Framework has a bug where it doesn't handle uncompressed files correctly.
+	// It writes them as "compressed" (DEFLATE) but with a compression level of 0. This causes
+	// issues with Android, which expect uncompressed files to be stored correctly.
+	// We can work around this by using LibZipSharp, which doesn't have this bug.
+	// This is only necessary if we're on .NET Framework (MSBuild in VSWin) and we have uncompressed files.
+	bool ShouldFallbackToLibZipSharp ()
 	{
-		var compressionMethod = GetCompressionMethod (item);
-		existingEntries.Remove (inArchivePath.Replace (Path.DirectorySeparatorChar, '/'));
+		// Explicitly requested via MSBuild property.
+		if (UseLibZipSharp) {
+			Log.LogDebugMessage ("Falling back to LibZipSharp because '$(_AndroidUseLibZipSharp)' is 'true'.");
+			return true;
+		}
+
+		// .NET 6+ handles uncompressed files correctly, so we don't need to fallback.
+		if (RuntimeInformation.FrameworkDescription == ".NET")
+			return false;
+
+		// Nothing is going to get written uncompressed, so we don't need to fallback.
+		if (uncompressedMethod != CompressionMethod.Store)
+			return false;
 
-		if (apk.SkipExistingFile (file, inArchivePath, compressionMethod)) {
-			Log.LogDebugMessage ($"Skipping {file} as the archive file is up to date.");
+		// No uncompressed file extensions were specified, so we don't need to fallback.
+		if (UncompressedFileExtensionsSet.Count == 0)
 			return false;
```

Comment:
In each of these cases, can we put a `LogDebugMessage()` that says why it returned `false` or `true`? Just thinking if `RuntimeInformation.FrameworkDescription` is unexpected, etc.